crates/core/tedge/src/cli/connect/error.rs

```diff
@@ -49,6 +49,9 @@ pub enum ConnectError {
     #[error("Could not parse private key")]
     RumqttcPrivateKey,
 
+    #[error("Unsupported key format")]
+    UnsupportedPvtKeyFormat,
+
```

Comment:
What is the difference between an `UnsupportedPvtKeyFormat` and a `RumqttcPrivateKey`?

* I would return an "unsupported key" error only if this is actually a private key but using a non-supported format.
* With the current code base, one can only say that the key file cannot be parsed. That's enough.

Hence, I would keep a single error case.

```suggestion
    #[error("Fail to parse the private key")]
    UnknownPrivateKeyFormat,
```

Reply:
Makes sense, now I have only one error